go/vt/vtgate/semantics/scoper.go

```diff
@@ -212,7 +211,7 @@ func (s *scoper) createSpecialScopePostProjection(parent sqlparser.SQLNode) erro
 			}
 			thisTableInfo := createVTableInfoForExpressions(sel.SelectExprs, nil /*needed for star expressions*/, s.org)
 			if len(tableInfo.cols) != len(thisTableInfo.cols) {
-				return engine.ErrWrongNumberOfColumnsInSelect
+				return vterrors.NewErrorf(vtrpcpb.Code_FAILED_PRECONDITION, vterrors.WrongNumberOfColumnsInSelect, "The used SELECT statements have a different number of columns")
```

Comment:
Since `engine` still depends on `semantics`, maybe we could / should move this error definition there so we can still use one?

Reply:
`engine` does not depend on `semantics` though? Or did you mean though the `engine`->`topsrv`->`topo`->`sidecardb`->`schemadif`->`semantics` path?

Reply:
@shlomi-noach Right, it's already indirectly depending on it so a direct dependency wouldn't make things worse?

Reply:
cc @harshit-gangal for authoritative opinion

Reply:
(in our pairing session @harshit-gangal suggested we do not mvoe the error to `semantics`)

Reply:
`engine` does not have a direct dependency on `semantics` for that reason I asked to avoid it. But if indirect dependency exists then I do not mind.

Reply:
It also feels like for a single error message we are creating a dependency which I felt is not required.